platforms/documentation/docs/src/docs/userguide/dep-man/declaring-depedencies/declaring_dependencies_basics.adoc

```diff
@@ -0,0 +1,221 @@
+
+[[sub:file_dependencies]]
+=== 3. File dependencies
+
+File dependencies allow you to include external JARs or other files directly into your project by referencing their file paths.
+
+NOTE: File dependencies are generally discouraged.
+Instead, prefer declaring dependencies on an external repository, or if necessary, declaring a maven or ivy repository using a `file://` URL.
+
+image::dependency-management-file-dependencies.png[]
+
+File dependencies are unique because they represent a direct reference to files on the filesystem without any associated <<glossary.adoc#sub:terminology_module_metadata,metadata>>, such as transitive dependencies, origin, or author information.
+
+====
+include::sample[dir="snippets/dependencyManagement/declaringDependencies-fileDependencies/kotlin",files="build.gradle.kts[tags=file-dependencies]"]
+include::sample[dir="snippets/dependencyManagement/declaringDependencies-fileDependencies/groovy",files="build.gradle[tags=file-dependencies]"]
+====
+
+In this example, each dependency explicitly specifies its location within the file system.
+Common methods for referencing these files include:
+
+- link:link:{groovyDslPath}/org.gradle.api.Project.html#org.gradle.api.Project:files(java.lang.Object++[]++)[`Project.files()`]: Accepts one or more file paths directly.
+- link:{javadocPath}/org/gradle/api/file/ProjectLayout.html#files-java.lang.Object++...++-[`ProjectLayout.files()`]: Accepts one or more file paths directly.
+- link:{groovyDslPath}/org.gradle.api.Project.html#org.gradle.api.Project:fileTree(java.lang.Object)[`Project.fileTree()`]: Defines a directory and includes or excludes specific file patterns.
+
+[NOTE]
+====
+The order of files in a `FileTree` is not guaranteed to be stable, even on the same computer.
+As a result, dependency configurations using `FileTree` may produce resolution results with varying order, which can impact the cacheability of tasks that use these results as inputs.
+To ensure more predictable and stable behavior, it is recommended to use the simpler `files` method where possible.
+====
+
+Alternatively, you can use a <<declaring_repositories_adv.adoc#sub:flat_dir_resolver,flat directory>> repository to specify the source directory for multiple file dependencies.
+
+Ideally, you should use Maven or Ivy repository with a local URL:
+
+[source,groovy]
+----
+repositories {
+    maven {
+        url 'file:///path/to/local/files' // Replace with your actual path
+    }
+}
+----
+
+For more details on working with file dependencies, refer to the link:{groovyDslPath}/org.gradle.api.artifacts.dsl.DependencyHandler.html[`DependencyHandler`] and `ProjectLayout` classes in the Gradle API documentation.
+
+File dependencies also allow you to add a set of files directly to a configuration without using a repository.
```

Comment:
💭 This sentence seems like it should go early, perhaps right after "File dependencies allow you to include external JARs or other files directly into your project by referencing their file paths."

Reply:
Moved.